src/net.cpp

```diff
@@ -692,29 +713,35 @@ const uint256& V1TransportDeserializer::GetMessageHash() const
     return data_hash;
 }
 
-CNetMessage V1TransportDeserializer::GetMessage(const CMessageHeader::MessageStartChars& message_start, int64_t time) {
+Optional<CNetMessage> V1TransportDeserializer::GetMessage(int64_t time) {
     // decompose a single CNetMessage from the TransportDeserializer
     CNetMessage msg(std::move(vRecv));
 
-    // store state about valid header, netmagic and checksum
-    msg.m_valid_header = hdr.IsValid(message_start);
-    msg.m_valid_netmagic = (memcmp(hdr.pchMessageStart, message_start, CMessageHeader::MESSAGE_START_SIZE) == 0);
-    uint256 hash = GetMessageHash();
+    // Check the header
+    if (!hdr.IsValid()){
```

Comment:
Seems like an error message here might be appropriate similar to the checksum check?

Reply:
The logging is done in IsValid and in ReadHeader.  I agree, I'd love to have it in GetMessage instead, just like the checksum check, but that would require IsValid to pass the information about why it failed up to GetMessage.  Ideally all of the logging would be in ReceiveMsgBytes (and indeed I had a version of this PR where that was the case) but seeing the complexity that added to Read, ReadHeader, IsValid, and GetHeader I didn't think it was worth it.